api.swagger.yaml

```diff
@@ -467,96 +576,803 @@ paths:
               schema:
                 $ref: '#/components/schemas/location-uuid-list-schema'
         '400':
-          description: The supplied UUID was malformed.
+          description: The supplied UUID was malformed.
+        '404':
+          description: The UUID does not correspond to an existing user.
+  '/users/{user_id}/locations/{location_id}':
+    delete:
+      tags:
+        - users
+      description: Remove a location from a user.
+      parameters:
+        - description: The Invotra UUID of the user.
+          in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+        - description: The Invotra UUID of the location.
+          in: path
+          name: location_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: The location has been removed from the user.
+        '400':
+          description: One or both of the UUIDs was malformed.
+        '404':
+          description: >
+            The user or location does not exist, or the user did not have that
+            location.
+    put:
+      tags:
+        - users
+      description: Add a location to a user.
+      parameters:
+        - description: The Invotra UUID of the user.
+          in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+        - description: The Invotra UUID of the job role.
+          in: path
+          name: location_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: The user already has this location.
+        '201':
+          description: The location was added to the user.
+        '400':
+          description: One or both of the UUIDs was malformed.
+        '404':
+          description: The user or location does not exist.
+  '/users/{user_id}/teams':
+    get:
+      tags:
+        - users
+      description: List a user's teams
+      parameters:
```

Comment:
I think this should probably have a pager too...

Reply:
added